src/EnergyPlus/General.cc

```diff
@@ -3656,22 +3656,19 @@ namespace General {
         return results;
     }
 
-    Real64 epexp(Real64 x)
+    Real64 epexp(Real64 numerator, Real64 denominator, Real64 overflowReturn, Real64 defaultHigh)
```

Comment:
@jasondegraw, the second function (with defaultHigh) is never used within the code (that I could find). I'd propose dropping that from the combined function. The rest of it always happens (or should always happen) anyway outside the function every time it's called to prevent the divide-by-zero, so why not bring it into the function?

There is a possibility of overflow still that we might want to protect against, but the current code isn't doing that (again, as far as I can tell).

Reply:
@nealkruis Used or not is not a good argument here, the original intent was (and still is) to begin building out functions with better protections and more options in terms of what is calculated, and that as that happened we'd start using these throughout the code. Requiring that every call to `exp`must involve a numerator and a denominator would run rather counter to the original goal. That's why moving the check inside is a bad idea. I'm not disputing that there are cases where there may need to be additional protections, that's all fine. The combined function is not.

Reply:
Okay, so maybe we'll wrap the call from Airflow Network in another function that checks the denominator and then calls `epexp`. That way you don't miss one of the cases with the external checks.

I still don't really understand the intent of the second form of the function. Why pass in a "defaultHigh"? There is a real limit on the arguments to exp before it returns HUGE_VAL (~370). If you want to avoid an invalid return, shouldn't we just limit the input to be max(x,370)?

Also, is there a reason to use -70 as the lower bound for tiny numbers? or is that a somewhat arbitrary cut off?

BTW, the suggested function did not require a denominator (you could just do `epexp(x)` still), but it would sometimes do an otherwise unnecessary divide by 1.0 (does the cpu even do an operation for divide by 1.0?).

Reply:
If I recall correctly, something just under -70 causes an underflow and over 370 an overflow. Those limits could easily be tested.

Reply:
Just tested: Underflow doesn't really happen. At some point, C++ just returns zero. Overflow in C++ is actually closer to 700.

Reply:
@nealkruis I think it's fine to have a form that has a denominator as an argument, there's precedent for that sort of thing, e.g. `atan2`. But there's still an `atan`, so no, wrapping an additional function around this is not a solution. All I asked was that we keep the two separate functions, one with a single argument and one with the additional arguments. It's really not that hard.

Reply:
As I remembered, the current epexp function deals with underflow only and will not process overflow. Since possible problems in AFN are for underflow, the current function should be enough. There is no need to add more arguments.

In addition, there is no need to remove "if (AirflowNetworkLinkSimu(i).FLOW2 == 0.0) {". If we know the solution at FLOW = 0, it is not needed to call the function epexp for speedup.

Reply:
@nealkruis Chiming in late in the game, but underflows are a thing, at least on Windows debug builds. I stumble across them frequently in the Tarcog functions. So, let's not remove underflow protection.

Reply:
@mjwitte My understanding is that `std::exp` handles underflow internally and returns zero.